src/main/java/duke/Duke.java

```
@@ -0,0 +1,233 @@
+import duke.exception.IllegalTodoException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Scanner;
+
+public class Duke {
+
+    /**
+     * Returns boolean value of true if input String is an integer,
+     * else returns boolean value of false
+     *
+     * @param word String input to check if it is an integer
+     * @return true if input String is an integer, otherwise false
+     */
+    public static boolean isNumeric(String word) {
+        int valueToConvert;
+        try {
+            valueToConvert = Integer.parseInt(word);
+            return true;
+        } catch (NumberFormatException e) {
+            System.out.println("Incorrect input, please enter an integer.");
+        }
+        return false;
+    }
+
+    public static void main(String[] args) {
+        greetingMessage();
+
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        Task[] tasks = new Task[100];
+        int taskCount = 0;
+
+        while (!line.equals("bye")) { // Exits the program if input is "bye"
+            String[] words = line.split(" ");
+            if (line.equals("list")) {
+                // List out all the tasks added
+                list(tasks, taskCount);
+            } else if (words[0].equals("unmark") && (words.length == 2) && (isNumeric(words[1]))) {
+                // Mark a task as not done
+                unmarkTask(tasks, taskCount, words);
+            } else if (words[0].equals("mark") && (words.length == 2) && (isNumeric(words[1]))) {
+                // Mark a task as done
+                markTask(tasks, taskCount, words);
+            } else {
+                // Adding a task to the list
+                try {
+                    addTask(line, tasks, taskCount);
+                    taskCount++;
+                } catch (IllegalTaskException e) {
+                    // Ignoring empty command inputs
```

Don't have empty catches